pkg/kubelet/kubelet.go

```diff
@@ -1088,6 +1089,7 @@ func (kl *Kubelet) syncPod(pod *api.BoundPod, dockerContainers dockertools.Docke
 				}
 				glog.Infof("pod %q container %q is unhealthy. Container will be killed and re-created.", podFullName, container.Name, live)
 			} else {
+				podChanged = true
```

Comment:
previously, we kill the infra pod container whenever there is a container change. This pull request does not change this behavior. The only change is to recreate the killed infra container after we finish the sync.